Classes/common/touchdb/TDBlobStore.m

```diff
@@ -198,17 +202,53 @@ - (NSInteger)deleteBlobsExceptWithKeys:(NSSet*)keysToKeep withDatabase:(FMDataba
         if ([[NSFileManager defaultManager] removeItemAtPath:blobPath error:&thisError]) {
             // Remove from db
             [TD_Database deleteRowForKey:oneRow.key inBlobFilenamesTableInDatabase:db];
-
-            ++numDeleted;
         } else {
             CDTLogError(CDTDATASTORE_LOG_CONTEXT, @"%@: Failed to delete '%@': %@", self,
                         oneRow.blobFilename, thisError);
 
-            errors = YES;
+            success = NO;
+            
+            // Do not try to delete it later, it will not be deleted from db
+            [filesToKeep addObject:oneRow.blobFilename];
         }
     }
 
-    return (errors ? -1 : numDeleted);
+    // Delete files not related to an attachments
+    [self deleteFilesExceptWithFilenames:filesToKeep];
+
+    // Return
+    return success;
+}
+
+- (void)deleteFilesExceptWithFilenames:(NSSet*)filesToKeep
+{
+    NSFileManager* defaultManager = [NSFileManager defaultManager];
+
+    // Read directory
+    NSError* thisError = nil;
+    NSArray* remainingFiles = [defaultManager contentsOfDirectoryAtPath:_path error:&thisError];
+    if (!remainingFiles) {
+        CDTLogError(CDTDATASTORE_LOG_CONTEXT, @"Can not read dir %@: %@", _path, thisError);
+
+        return;
+    }
+
+    // Delete all files but exceptions
+    for (NSString* oneFilename in remainingFiles) {
```

Comment:
Remove the `one` prefix where a singular would do, `oneFilename` is the same as `filename`?

Also, I can see you named `remainingFiles` to be "remaining after deleting the other ones in the method that calls this one", however, if you only consider the context of this method, the name doesn't make sense (remaining files implies remaining after an operation, but getting the filenames is the first thing that this method does). I suggest `currentFiles` or just `filenames`.

I'd also be tempted to pass the path into the method, gives you a nicer name: `deleteFilesNotInSet:fromPath:`.